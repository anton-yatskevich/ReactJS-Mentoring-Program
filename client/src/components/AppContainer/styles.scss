* {
	margin: 0;
	padding: 0;
	box-sizing: border-box;
}

body {
	background-color: rgb(230, 230, 230);
	font-family: Arial, Helvetica, sans-serif;
	font-size: 16px;
}


