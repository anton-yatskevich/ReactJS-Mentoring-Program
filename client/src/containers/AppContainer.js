import React, { Component } from 'react';
import Foooter from '../components/Footer';
import Header from '../components/Header';
import SearchPage from './SearchPage';
import MOVIES_DATA from '../constants/mockData.json';
import MoviePage from '../components/MoviePage';
import ErrorBoundary from './ErrorBoundary';

class AppContainer extends Component {
    constructor() {
        super();
        this.movies = MOVIES_DATA;
        this.onSelectMovie = this.onSelectMovie.bind(this);
        this.goToSearchHandler = this.goToSearchHandler.bind(this);
        this.state = {
            selectedMovie: null
        };
    }

    onSelectMovie(id) {
        const movie = this.movies.filter(item => item.id === id)[0];
        this.setState({
            selectedMovie: movie
        });
    }

    goToSearchHandler() {
        this.setState({
            selectedMovie: null
        });
    }

    render() {
        const { selectedMovie } = this.state;

        return (
            <ErrorBoundary>
                <Header
                    isSearchPage={selectedMovie === null}
                    goToSearchHandler={this.goToSearchHandler}
                />
                <main>
                    {
                        selectedMovie
                            ? (
                                <MoviePage
                                    movies={this.movies}
                                    selectedMovie={selectedMovie}
                                    onSelectMovie={this.onSelectMovie}
                                />
                            )
                            : <SearchPage movies={this.movies} onSelectMovie={this.onSelectMovie} />
                    }
                </main>
                <Foooter />
            </ErrorBoundary>
        );
    }
}

export default AppContainer;
