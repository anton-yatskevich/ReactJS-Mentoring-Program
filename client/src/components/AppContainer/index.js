import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import noop from 'lodash.noop';
import Header from '../Header';
import SearchPage from '../SearchPage';
import MoviePage from '../../containers/MoviePageContainer';
import MovieSchema from '../../constants/MovieSchema';
import NotFound from '../NotFound';
import './styles.scss';

const AppContainer = ({ selectedMovie, selectMovie, numberOfResults }) => (
    <Router>
        <Header
            isSearchPage={selectedMovie === null}
            goToSearchHandler={selectMovie}
        />
        <main>
            <Switch>
                <Route path="/" exact component={() => <SearchPage numberOfResults={numberOfResults} />} />
                <Route path="/film/:id" component={MoviePage} />
                <Route path="" component={NotFound} />
            </Switch>
        </main>
        <footer>Copyright Â© 2019</footer>
    </Router>
);

AppContainer.propTypes = {
    numberOfResults: PropTypes.number,
    selectedMovie: MovieSchema,
    selectMovie: PropTypes.func
};

AppContainer.defaultProps = {
    numberOfResults: 0,
    selectedMovie: {},
    selectMovie: noop
};


export default AppContainer;
