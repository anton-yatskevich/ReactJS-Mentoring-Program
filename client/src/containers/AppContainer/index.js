import React, { Component } from 'react';
import Header from '../../components/Header';
import SearchPage from '../SearchPage';
import MOVIES_DATA from '../../constants/mockData.json';
import MoviePage from '../../components/MoviePage';
import ErrorHandlerComponent from '../ErrorBoundary';
import './styles.scss';

class AppContainer extends Component {
    state = {
        selectedMovie: null
    };

    movies = MOVIES_DATA;

    onSelectMovie = (id) => {
        const movie = this.movies.find(item => item.id === id);
        this.setState({
            selectedMovie: movie
        });
    }

    goToSearchHandler = () => {
        this.setState({
            selectedMovie: null
        });
    }

    render() {
        const { selectedMovie } = this.state;

        return (
            <>
                <Header
                    isSearchPage={selectedMovie === null}
                    goToSearchHandler={this.goToSearchHandler}
                />
                <main>
                    {
                        selectedMovie
                            ? (
                                <MoviePage
                                    movies={this.movies}
                                    selectedMovie={selectedMovie}
                                    onSelectMovie={this.onSelectMovie}
                                />
                            )
                            : <SearchPage movies={this.movies} onSelectMovie={this.onSelectMovie} />
                    }
                </main>
                <footer>Copyright Â© 2019</footer>
            </>
        );
    }
}

export default ErrorHandlerComponent(AppContainer);
