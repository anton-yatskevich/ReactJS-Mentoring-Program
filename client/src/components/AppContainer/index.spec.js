import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AppContainer from './index';

describe('AppContainer component', () => {
    let AppContainerComponent;

    beforeAll(() => {
        AppContainerComponent = shallow(<AppContainer />);
    });

    it('should be rendered with default props', () => {
        expect(toJson(AppContainerComponent)).toMatchSnapshot();
    });
});
