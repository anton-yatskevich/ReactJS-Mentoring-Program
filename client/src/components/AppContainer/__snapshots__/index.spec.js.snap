// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer component should be rendered with default props 1`] = `
<Fragment>
  <Header
    goToSearchHandler={[Function]}
    isSearchPage={false}
  />
  <main>
    <Connect(MoviePage) />
  </main>
  <footer>
    Copyright Â© 2019
  </footer>
</Fragment>
`;
